UMN-Common.psm1

```diff
@@ -736,3 +736,109 @@ function Write-Log {
 		if ($Console -eq $true) {Write-Host $Message}
 
 } #END Write-Log
+
+function Get-RandomCharacter{
+	<#
+	.SYNOPSIS
+	This function is to randomize a list of characters for password generation.
+
+	.PARAMETER length
+	How many of each character set you want.
+
+	.PARAMETER characters
+	A set list of characters to be selected from.
+
+	.EXAMPLE
+	Get-RandomCharacters -length 2 -characters '1234567890'
+	42
+	#>
+
+	param(
+	[Parameter(Mandatory=$true)]$length,
+	[Parameter(Mandatory=$true)]$characters
+	)
+	Begin{}
+	Process{
+		$random = 1..$length | ForEach-Object {Get-Random -Maximum $characters.length}
+		$private:ofs=""
+	}
+	End{
+		return [String]$characters[$random]
+	}
+}
+
+function Scramble-String{
+	<#
+		.SYNOPSIS
+		This function takes a list of characters and randomizes them.
+
+		.PARAMETER inputString
+		A string of characters to scramble.
+
+		.EXAMPLE
+		Scramble-String -inputString '12345'
+		51432
+	#>
+	param(
+	[Parameter(Mandatory=$true)]
+	[string]$inputString
+	)
+	Begin{$characterArray = $inputString.ToCharArray()}
+	Process{
+		$scrambledStringArray = $characterArray | Get-Random -Count $characterArray.Length
+		$outputString = -join $scrambledStringArray
+	}
+	End{return $outputString}
+}
+
+function new-password{
+	<#
+		.SYNOPSIS
+		This function just generates a password based on predetermined character list.
+
+		.PARAMETER passwordLength
+		The length of the password
+
+		.EXAMPLE
+		new-password -passwordLength 15
+		42
+	#>
+	param(
+		[int]$passwordLength = 40
+	)
+	Begin{
+		$length = [System.Math]::Round(($length = $passwordLength/4),0)
```

Comment:
I think the intent here is to make sure you always get exactly as many characters as requested, even if the number isn't divisible by 4.  I simplified it quite a bit, using [Math]::Truncate instead of doing string operations, and putting it into a much simpler switch statement, taking advantage of the modulus (`%`) operator:
```powershell
[int]$length = [Math]::Truncate($passwordLength / 4)

switch ($passwordLength % 4) {
    0 { $lengths = @($length, $length, $length, $length) }
    1 { $lengths = @($length, $length, $length, ($length + 1)) }
    2 { $lengths = @($length, $length, ($length + 1), ($length + 1)) }
    3 { $lengths = @($length, ($length + 1), ($length + 1), ($length + 1)) }
}
```

Then at the bottom you can use `$lengths[0]` and `$lengths[1]` etc instead of `$length` and `$length2` etc

Reply:
Pretty - yeah - a lot cleaner.